lib/Core/AddressSpace.h

```diff
@@ -100,6 +111,19 @@ namespace klee {
                  unsigned maxResolutions=0,
                  time::Span timeout=time::Span()) const;
 
+    /// Resolve range `[begin, end)` to a list of `ObjectPairs` that
+    /// could overlap that range. If `maxResolutions` is non-zero then
+    /// no more than that many pairs will be returned.
+    ///
+    /// \return true iff the resolution is incomplete (`maxResolutions`
+    /// is non-zero and it was reached, or a query timed out).
+    bool resolveRange(ExecutionState &state,
```

Comment:
Why is this necessary? It seems when you invoke this, the range is within a cacheline or something. It's not clear to me why you need this.

Reply:
I explained this in the commit message, but I should add a comment too:

Some programs round their pointers to be cache-aligned
when flushing modifications to pmem. This means that it's
possible for the address passed to clwb/clflush to *not*
resolve to any known MemoryObject:
```
    Known MemoryObject: 0x10008, size 8
    Program calls clwb(0x10000)
```

Solution: rather than searching for MemoryObjects that contain a single
pointer, resolve all MemoryObjects that overlap any part of the cache
line being flushed. New function `AddressSpace::resolveRange` added to
help achieve this. Also led to more complicated executeFlush logic.

Reply:
Gotcha. Yeah put this in a comment, it's easier than git logging for documentation.

It actually seems like we should be using this for stores to persistent memory as well (especially for non-temporals), as you could construct a single store instruction that could dirty more than one cache line (i.e., a vector write of two 64-bit ints could store one at the end of one cache line and the next at the beginning of another cache line.)

Reply:
Address resolution already works properly for loads and stores.

Reply:
Oh right, because it ends up calling write8 a bunch.